app-arch/libarchive/libarchive-3.2.1-r6.ebuild

```diff
@@ -12,25 +12,28 @@ SRC_URI="http://www.libarchive.org/downloads/${P}.tar.gz"
 LICENSE="BSD BSD-2 BSD-4 public-domain"
 SLOT="0/13"
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-fbsd ~sparc-fbsd ~x86-fbsd ~x86-interix ~amd64-linux ~arm-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~m68k-mint ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
-IUSE="acl +bzip2 +e2fsprogs expat +iconv kernel_linux libressl lz4 +lzma lzo nettle static-libs +threads xattr +zlib"
-
-RDEPEND="
-	acl? ( virtual/acl[${MULTILIB_USEDEP}] )
-	bzip2? ( app-arch/bzip2[${MULTILIB_USEDEP}] )
-	expat? ( dev-libs/expat[${MULTILIB_USEDEP}] )
-	!expat? ( dev-libs/libxml2[${MULTILIB_USEDEP}] )
-	iconv? ( virtual/libiconv[${MULTILIB_USEDEP}] )
+IUSE="acl +bzip2 +e2fsprogs expat +iconv kernel_linux libressl lz4 +lzma lzo nettle static static-libs +threads xattr +zlib"
+REQUIRED_USE="static? ( static-libs )"
+
+LIB_DEPEND="
+	acl? ( virtual/acl[${MULTILIB_USEDEP},static-libs(+)] )
+	bzip2? ( app-arch/bzip2[${MULTILIB_USEDEP},static-libs(+)] )
+	expat? ( dev-libs/expat[${MULTILIB_USEDEP},static-libs(+)] )
+	!expat? ( dev-libs/libxml2[${MULTILIB_USEDEP},static-libs(+)] )
+	iconv? ( virtual/libiconv[${MULTILIB_USEDEP},static-libs(+)] )
 	kernel_linux? (
-		xattr? ( sys-apps/attr[${MULTILIB_USEDEP}] )
+		xattr? ( sys-apps/attr[${MULTILIB_USEDEP},static-libs(+)] )
 	)
-	!libressl? ( dev-libs/openssl:0=[${MULTILIB_USEDEP}] )
-	libressl? ( dev-libs/libressl:0=[${MULTILIB_USEDEP}] )
-	lz4? ( >=app-arch/lz4-0_p131:0=[${MULTILIB_USEDEP}] )
-	lzma? ( app-arch/xz-utils[threads=,${MULTILIB_USEDEP}] )
-	lzo? ( >=dev-libs/lzo-2[${MULTILIB_USEDEP}] )
-	nettle? ( dev-libs/nettle:0=[${MULTILIB_USEDEP}] )
-	zlib? ( sys-libs/zlib[${MULTILIB_USEDEP}] )"
+	!libressl? ( dev-libs/openssl:0=[${MULTILIB_USEDEP},static-libs(+)] )
+	libressl? ( dev-libs/libressl:0=[${MULTILIB_USEDEP},static-libs(+)] )
+	lz4? ( >=app-arch/lz4-0_p131:0=[${MULTILIB_USEDEP},static-libs(+)] )
+	lzma? ( app-arch/xz-utils[threads=,${MULTILIB_USEDEP},static-libs(+)] )
+	lzo? ( >=dev-libs/lzo-2[${MULTILIB_USEDEP},static-libs(+)] )
+	nettle? ( dev-libs/nettle:0=[${MULTILIB_USEDEP},static-libs(+)] )
+	zlib? ( sys-libs/zlib[${MULTILIB_USEDEP},static-libs(+)] )"
+RDEPEND="!static? ( ${LIB_DEPEND//\[static-libs(+)]} )"
```

Comment:
instead of this replacement thing, just do `sys-libs/zlib[${MULTILIB_USEDEP},static-libs(+)?]` for instance. Many ebuilds in the tree do this. Doing bash replacement for `RDEPEND` is not nice.